Meme/Meme/ViewModel.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ViewModel.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/30/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+class ViewModel {
```

Comment:
Took a stab at a viewModel that initalizes with Network client @vhart . Let me know if this is not what you had in mind.

Reply:
You mean you don't like this generic name? Jk, sounds good!

Reply:
Could you elaborate a bit more on the purpose of the protocol for the VM? Is it to declare `requestTemplateExample`, `requestTemplates` and `requestMeme` and have the VM implement them in an extension?

And inject the protocol inside the VC instead of a Meme instance? Or is it for another reason?

Reply:
Also, I'm struggling on a good name for the VM, MemeViewModel seems a bit misleading since currently it's fetching templates and template examples as well. Any suggestions for a good name?

Reply:
I think the name will come if the vc has a clear name. This is for the, maybe `MemeTemplatesViewController`? In which case we'd call the vm `MemeTemplatesViewModel`
The protocol is to make the view model injectable. So for example if we had a vm that hit the network and another that used like a cache or file on disk we'd be able to swap them out as needed. Possibly overkill for the first iteration, but it's the ideal design for modularity.